client/src/components/AddProduct.js

```diff
@@ -0,0 +1,249 @@
+    const classes = useStyles()
+    const [title, setTitle]=useState('');
+    const [list, setList]=useState('');
+    const [errors, setErrors]=useState({});
+
+    const handleChange = (event) => {
+      setList(event.target.value);
+    };
+
+    const handleClose = () => {
+        changeAddProductOpen();
+    }
+
+    const validations = () => {
+        const errorsCopy = {...errors};
+        errorsCopy.title = title ? "" : "This field is required.";
+        setErrors({ ...errorsCopy });
+
+        return Object.values(errorsCopy).every(field => field === "");
+    }
+
+    const handleClick = (e) => {
+        e.preventDefault();
+        if (validations()) {
+            fetch("route", {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({
+                    'title': title,
+                    //'cover': cover
+                })})
+                .then(response => response.json())
+                .then(function(response) {
+                    if (response.status === 'success') {
+                        console.log('Success:');
+                    }
+                    else {
+                        window.alert(response.message); // Replace with snackbar.
+                        console.log(response.message);
+                    }
+                })
+                .catch((error) => {
+                    console.error('Error:', error);
+                })
+            }
+    }
+
+    return (
+        <Modal
+        className={classes.popup}
+        open={addProductOpen}
+        onClose={handleClose}
+        aria-labelledby="add-product"
+        aria-describedby="confirms-adding-item-of-url-to-selected-list"
+        >
+        <Box className={classes.paper}>
+          <Box className={classes.closeButtonContainer}>
+            <IconButton
+              aria-label="close"
+              className={classes.closeButton}
+              onClick={handleClose}
+              size="small"
+            >
+              <CloseIcon fontSize="small" />
+            </IconButton>
+          </Box>
+          <Box className={classes.titleContainer}>
+            <Typography className={classes.titleText}>Add new item:</Typography>
+          </Box>
+          <Box className={classes.bodyContainer}>
+            <Typography className={classes.itemText}>Paste link to item:</Typography>
+          </Box>
+          <Box className={classes.textFieldContainer}>
+            <TextField
+              className={classes.textField}
+              variant="outlined"
+              placeholder="Paste your link here" // Placeholder needs to be centered.
+              fullWidth
+              type="text"
+              error={!!errors.title}
+              helperText={errors.title}
+              onChange={(e)=>setTitle(e.target.value)}/>
```

Comment:
This field is for the link, right? Maybe a variable name of 'url' would be more clear than 'title'?